eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -137,13 +137,16 @@ jobs:
 
     - pwsh: |
         $toxenvvar = "whl,sdist"
-        if (('$(Build.Reason)' -eq 'Schedule') -and ('$(System.TeamProject)' -eq 'internal')) {
+        if ('$(System.TeamProject)' -eq 'internal') {
           $toxenvvar = "whl,sdist,depends,latestdependency,mindependency"
         }
 
-        if ('$(Run.DependencyTest)' -eq 'true') {
-          $toxenvvar = "whl,sdist,depends,latestdependency,mindependency"
+        # ensure that the variable is unset. if it isn't, use the value discovered there
+        if ('$(Run.ToxCustomEnvs)' -ne ('$' + '(Run.ToxCustomEnvs)'))
```

Comment:
LOL... I love this check :)

For other similar checks for variable existence (that aren't secrets) you can use the environment var instead. So `$env:RUN_TOXCUSTOMEVS` and it will be empty if not set or empty otherwise it will have the value.